website/docs/core-concepts/connecting-to-data-sources/authentication/connect-to-apis.md

```diff
@@ -4,24 +4,102 @@ description: You can hit any REST endpoint available on the public internet
+
+It's important to choose the appropriate method based on the task you want to accomplish, as each method has a specific meaning and can behave differently depending on the API you are using.
+
+
+### URL
+
+The URL Path provides a way to access the API and its functions. To use this field, you can enter the API URL that you want to access. For example, if you want to access the Appsmith mock API, you can enter the URL as `https://mock-api.appsmith.com`.
+
+### Headers
+
+HTTP headers are a critical part of the API request and response, as they contain meta-data about the API request and response. Headers carry the following information:
+
+* Request and Response Body;
+* Request Authorization;
+* Response Caching;
+* Response Cookies.
+
+It's important to be familiar with the major request and response headers, as you may need to set request headers when testing an API, and you may need to set assertions against response headers to ensure that the correct headers are being returned.
+
+### Params
+
+In a REST API, parameters are values that are passed in the URL, query string, or request body that are used to filter, sort, paginate, or perform some other operation on a set of data. These parameters allow the API to return a specific set of data based on the values that are provided.
+
+For example, an API might have a parameter called "limit" that specifies the maximum number of results to return, or a parameter called "sort" that determines the order in which the results should be sorted. By providing different values for these parameters, a client can control the data that's returned by the API.
+
+### Body
+
+In an API query, the body is the main content of the request. It's typically used to send data to the server, such as form data or a JSON payload. The body of an API request is usually included with POST and PUT requests, which are used to create or update resources on the server. For example, it might contain a JSON object with multiple key-value pairs, or it might contain a simple string or number.
+
+### Pagination
+
+Pagination refers to the process of receiving portions of a huge dataset until you receive the entire dataset. This helps optimize your app's performance because trying to get all the data results in a slow load time of your app and a slow rendering of the results in the UI. Appsmith supports the following methods for paginating API queries:
+
+
+#### Paginate with table page number
+Allows you to paginate the results of an API query using a table page number. To use this feature, you need to configure table and request parameters.
+
+This can be done by setting up request parameters to control the table's pagination. This may involve mapping a key like "**pageNo**" to the table's page number property.
+
+Example - Map key pageNo or similar to value
```

Comment:
While this may work out to be correct in certain cases, we should also provide one more example where we may want to configure `limit` and `offset` property separately. e.g. `limit = {{UsersTable.pageSize}}` and `offset={{UsersTable.pageOffset}}`

Reply:
done